src/Symfony/Component/Messenger/Transport/Receiver/QueueReceiverInterface.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Receiver;
+
+use Symfony\Component\Messenger\Envelope;
+
+/**
+ * Some transports may have multiple queues. This interface is used to read from only some queues.
+ *
+ * @author David Buchmann <mail@davidbu.ch>
+ */
+interface QueueReceiverInterface
```

Comment:
Should this extend the `ReceiverInterface`?

Reply:
i think so, yes. getting a message from a queue but not being able to ack/reject makes no sense. AmqpReceiver implemented both QueueReceiverInterface and ReceiverInterface, which is weird. i made this extend ReceiverInterface and simplified AmqpReceiver `implements`